### PR TITLE
dnfdaemon: Allow configuration overrides for root

### PR DESCRIPTION
**Pull request by m-blaha**
Root user is allowed to override all config options.

### CODE REVIEW

File: dnf5daemon-server/session.cpp [hunk 1]
```diff
@@ -245,11 +257,10 @@ bool Session::check_authorization(const std::string & actionid, const std::strin
     sdbus::Struct<bool, bool, std::map<std::string, std::string>> auth_result;
     sdbus::Struct<std::string, dnfdaemon::KeyValueMap> subject{"system-bus-name", {{"name", sender}}};
     std::map<std::string, std::string> details{};
-    uint flags = ALLOW_USER_INTERACTION;
     std::string cancelation_id = "";
     polkit_proxy->callMethod("CheckAuthorization")
         .onInterface(interface_name)
-        .withArguments(subject, actionid, details, flags, cancelation_id)
+        .withArguments(subject, actionid, details, allow_user_interaction ? ALLOW_USER_INTERACTION : 0, cancelation_id)
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/session.cpp:263]: Yeah, looks like conversion false->0 and true->1 is part of c++ standard. I'd like to make it explicit though, so I'll go with `allow_user_interaction ? 1 : 0`.

[the remark this answers is not in the document]
Comment (inknos) [reply dnf5daemon-server/session.cpp]: I was about to suggest a comment over it in the case you would just use the c++ boolean promotion to integer, but I like the explicit operation.

[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/session.cpp]: Sorry, converted PR back to draft. Found an issue here - the flags need to be `uint`.

[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5daemon-server/session.cpp]: OK, fixed, ready for review again.

**Comment by inknos** [comment]
there is a failing test due to this commit which does not impact this PR. https://github.com/rpm-software-management/ci-dnf-stack/commit/f280f0fab5beb4fddcdaf553ba4d9938ecd7412a

It looks good and the test should be fixed on another pr https://github.com/rpm-software-management/ci-dnf-stack/pull/1453

**Pull request merged**